test/common/gitops_utils.go

```diff
@@ -0,0 +1,167 @@
+
+	ocpUser := OCPUser{
+		ClusterRoles: []types.NamespacedName{
+			{Name: "open-cluster-management:admin:local-cluster"},
+			{
+				Name:      "admin",
+				Namespace: namespace,
+			},
+		},
+		ClusterRoleBindings: []string{subAdminBinding},
+		Password:            "",
+		Username:            username,
+	}
+
+	// Append any additional provided ClusterRoles
+	ocpUser.ClusterRoles = append(ocpUser.ClusterRoles, additionalRoles...)
+
+	// Occasionally, the subscription-admin ClusterRoleBinding may not exist due to some unknown
+	// error. This ClusterRoleBinding is supposed to have been created by the App Lifecycle
+	// controllers. In this unusual case, create the ClusterRoleBinding based on the advice from
+	// the Application Lifecycle squad.
+	subAdminBindingObj := rbacv1.ClusterRoleBinding{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: subAdminBinding,
+		},
+		RoleRef: rbacv1.RoleRef{
+			APIGroup: "rbac.authorization.k8s.io",
+			Kind:     "ClusterRole",
+			Name:     subAdminBinding,
+		},
+	}
+
+	By("Verifying that the subscription-admin ClusterRoleBinding exists")
+
+	_, err := ClientHub.RbacV1().ClusterRoleBindings().Create(
+		context.TODO(), &subAdminBindingObj, metav1.CreateOptions{},
+	)
+	if err != nil {
+		Expect(k8serrors.IsAlreadyExists(err)).Should(
+			BeTrue(),
+			"Expected error to be 'already exists': "+fmt.Sprint(err),
+		)
+	}
+
+	By("Cleaning up any existing subscription-admin user config")
+	GitOpsCleanup(namespace, "", ocpUser)
+
+	By("Creating a subscription-admin user and configuring IDP")
+	// Create a namespace to house the subscription configuration.
+	nsObj := corev1.Namespace{ObjectMeta: metav1.ObjectMeta{Name: namespace}}
+	_, err = ClientHub.CoreV1().Namespaces().Create(
+		context.TODO(), &nsObj, metav1.CreateOptions{},
+	)
+	Expect(err).Should(BeNil())
+
+	// Create a subscription and local-cluster administrator OpenShift user that can be used
+	// for logging in.
+	ocpUser.Password, err = GenerateInsecurePassword()
+	Expect(err).Should(BeNil())
+
+	err = CreateOCPUser(ClientHub, ClientHubDynamic, ocpUser)
+	Expect(err).Should(BeNil())
+
+	// Get a kubeconfig logged in as the subscription and local-cluster administrator OpenShift
+	// user.
+	hubServerURL, err := OcHub("whoami", "--show-server=true")
+	Expect(err).Should(BeNil())
+
+	hubServerURL = strings.TrimSuffix(hubServerURL, "\n")
+	// Use eventually since it can take a while for OpenShift to configure itself with the new
+	// identity provider (IDP).
+	const fiveMinutes = 5 * 60
+	var kubeconfigSubAdmin string
+
+	Eventually(
+		func() error {
+			var err error
+			kubeconfigSubAdmin, err = GetKubeConfig(
+				hubServerURL, ocpUser.Username, ocpUser.Password,
+			)
+
+			return err
+		},
+		fiveMinutes,
+		10,
+	).Should(BeNil())
+
+	return ocpUser, kubeconfigSubAdmin
+}
+
+func GitOpsDeployFile(path string, namespace string, kubeconfig string) {
+	_, err := OcHub(
+		"apply",
+		"-f",
+		path,
+		"-n",
+		namespace,
+		"--kubeconfig="+kubeconfig,
```

Comment:
I don't think this will work because this is calling `OcHub` which appends the hub kubeconfig at the end of the arguments.

Reply:
Oh my goodness. That could be the source of the flakiness! This was definitely copy/paste:
https://github.com/stolostron/governance-policy-framework/blob/46892b5e56e25018c191e6609dc3560cb5967444/test/integration/policy_generator_test.go#L113
https://github.com/stolostron/governance-policy-framework/blob/46892b5e56e25018c191e6609dc3560cb5967444/test/integration/policy_generator_acm_hardening_test.go#L176